processor/src/it/java/com/linecorp/decaton/processor/ArbitraryTopicTypeTest.java

```diff
@@ -0,0 +1,166 @@
+            final Properties properties = TestUtils.defaultProducerProps(bootstrapServers);
+
+            this.topic = topic;
+            producer = new KafkaProducer<>(properties, keySerializer, valueSerializer);
+        }
+
+        public CompletableFuture<RecordMetadata> put(K key, V value) {
+            final ProducerRecord<K, V> record = new ProducerRecord<>(topic, key, value);
+
+            final CompletableFuture<RecordMetadata> result = new CompletableFuture<>();
+            producer.send(record, (metadata, exception) -> {
+                if (exception == null) {
+                    result.complete(metadata);
+                } else {
+                    result.completeExceptionally(exception);
+                }
+            });
+
+            return result;
+        }
+
+        @Override
+        public void close() throws Exception {
+            producer.close();
+        }
+    }
+
+    private static final class TestTaskExtractor<T> implements TaskExtractor<T> {
+        private final String topic;
+        private final Deserializer<T> deserializer;
+
+        private TestTaskExtractor(String topic, Deserializer<T> deserializer) {
+            this.topic = topic;
+            this.deserializer = deserializer;
+        }
+
+        @Override
+        public DecatonTask<T> extract(byte[] bytes) {
+            final T value = deserializer.deserialize(topic, bytes);
+            final TaskMetadata metadata = TaskMetadata.builder().build();
+            return new DecatonTask<>(metadata, value, bytes);
+        }
+    }
+
+    private <K, V> void testRetryWithKeyValue(
+            Serializer<K> keySerializer,
+            K key,
+            Serializer<V> valueSerializer,
+            Deserializer<V> valueDeserializer,
+            V value
+    ) throws Exception {
+        final CountDownLatch processLatch = new CountDownLatch(1);
+        final RetryConfig retryConfig = RetryConfig.builder().retryTopic(retryTopic).backoff(Duration.ofMillis(10)).build();
+        final Consumer<SubscriptionBuilder> builderConfigurer = builder -> builder.processorsBuilder(
+                ProcessorsBuilder.consuming(topic, new TestTaskExtractor<>(topic, valueDeserializer)).thenProcess((context, task) -> {
+                    if (context.metadata().retryCount() == 0) {
+                        context.retry();
+                    } else {
+                        processLatch.countDown();
+                    }
+                })).enableRetry(retryConfig);
+
+        try (ProcessorSubscription subscription = TestUtils.subscription(rule.bootstrapServers(), builderConfigurer);
+             TestProducer<K, V> producer = new TestProducer<>(rule.bootstrapServers(), topic, keySerializer, valueSerializer)) {
+            producer.put(key, value);
+            processLatch.await();
+        }
+    }
+
+    @Test(timeout = 30000)
+    public void testPrintableAsciiStringKeyValue() throws Exception {
```

Comment:
I guess String-key based scenario is already tested in other basic tests.
The intention to have another test here is?

Reply:
I just thought that there were no test scenarios that produce String-key records without `DecatonClient` but an arbitrary `Producer`.

But it looks we already have such scenarios? Let me consider removing this one then

Reply:
https://github.com/line/decaton/pull/157/commits/d79abd2662000ccb36ebcf2b65e4153f70e95154 Replaced with byte key/value test case (or we don't need two tests?)